src/bicep/mlz.bicep

```diff
@@ -1,30 +1,457 @@
 targetScope = 'subscription'
 
+/*
+
+  PARAMETERS
+
+  Here are all the parameters a user can override.
+
+  These are the required parameters that Mission LZ does not provide a default for:
+    - resourcePrefix
+
+*/
+
+// REQUIRED PARAMETERS
+
+@minLength(3)
+@maxLength(10)
+@description('A prefix, 3-10 alphanumeric characters without whitespace, used to prefix resources and generate uniqueness for resources with globally unique naming requirements like Storage Accounts and Log Analytics Workspaces')
+param resourcePrefix string
+
+@description('The subscription ID for the Hub Network and resources. It defaults to the deployment subscription.')
+param hubSubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Identity Network and resources. It defaults to the deployment subscription.')
+param identitySubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Operations Network and resources. It defaults to the deployment subscription.')
+param operationsSubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Shared Services Network and resources. It defaults to the deployment subscription.')
+param sharedServicesSubscriptionId string = subscription().subscriptionId
+
+@description('The region to deploy resources into. It defaults to the deployment location.')
+param location string = deployment().location
+
+// RESOURCE NAMING PARAMETERS
+
+@minLength(3)
+@maxLength(6)
+@description('A suffix, 3 to 6 characters in length, to append to resource names (e.g. "dev", "test", "prod", "mlz"). It defaults to "mlz".')
+param resourceSuffix string = 'mlz'
+
+@description('A timestamp to use for naming deployments uniquely. It defaults to the Bicep resolution of the "utcNow()" function.')
+param nowUtc string = utcNow()
```

Comment:
The name of this parameter does not make its usage obvious. It is used only for deployment names, so how about renaming it to `deploymentNameSuffix` or `deploymentNameSeed`? This could be done in a separate PR because renaming parameters is not in the defined scope of this work.

Reply:
Renamed to `deploymentNameSuffix`, I like that. Thanks for the suggestion. Helpful that we no longer use `utcNow()` in any assignment.